### PR TITLE
chore: Get up to date *again*

### PR DESCRIPTION
**Pull request by vladfrangu**
**Please describe the changes this PR makes and why it should be merged:**

~~Draft for now, as I still have lots to do :megu:~~

- Updates dependencies
- Cleans up message buttons, makes them stricter too
- Brings some oauth2 docs up tod ate
- Corrects the key for the `nick` change, it's now ACTUALLY `nick` instead of what it was previously
- Adds the missing `application_id` property to messages
- Fixes the unversioned API bases in rest exports
- Adds all new guild features to v9 (and v8 once @advaith1 and I talk)
- Fixes the rudimentary interactions types, making them stricter and cleaner
- If you're like me and want to test your unions, tsd is now present, so have at it!

**Reference Discord API Docs PRs or commits:**

### CODE REVIEW

**Review by kyranet — APPROVED** [review]

File: payloads/v9/_interactions/responses.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	 */
+	DeferredMessageUpdate,
+	/**
+	 * ACK a button interaction and edit the message to which the button was attached
+	 */
+	UpdateMessage,
+}
+
+/**
+ * https://discord.com/developers/docs/interactions/slash-commands#interaction-response-interactionapplicationcommandcallbackdata
```
Comment (monbrey) [inline payloads/v9/_interactions/responses.ts:73]:
```suggestion
 * https://discord.com/developers/docs/interactions/slash-commands#interaction-response-object-interaction-application-command-callback-data-structure
```

File: payloads/v9/_interactions/slashCommands.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
+import { APIRole, APIUser } from '../index';
+import { Permissions, Snowflake } from '../../../globals';
+import { APIPartialChannel } from '../channel';
+import { APIGuildMember } from '../guild';
+import { APIBaseInteraction } from './base';
+import { InteractionType } from './responses';
+
+/**
+ * https://discord.com/developers/docs/interactions/slash-commands#applicationcommand
```
Comment (monbrey) [inline payloads/v9/_interactions/slashCommands.ts:9]:
```suggestion
 * https://discord.com/developers/docs/interactions/slash-commands#application-command-object
```

File: payloads/v9/_interactions/slashCommands.ts [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+ */
+export interface APIApplicationCommand {
+	/**
+	 * Unique id of the command
+	 */
+	id: Snowflake;
+	/**
+	 * Unique id of the parent application
+	 */
+	application_id: Snowflake;
```
Comment (monbrey) [inline payloads/v9/_interactions/slashCommands.ts:19]:
```suggestion
	application_id: Snowflake;
	/**
	 * Guild id of the command, if not global
	 */
	guild_id?: Snowflake;
```
Unsure if this belongs here but it was missing

File: payloads/v9/_interactions/slashCommands.ts [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+		| ApplicationCommandOptionType.Role
+		| ApplicationCommandOptionType.Mentionable;
+	name: string;
+	description: string;
+	default?: boolean;
+	required?: boolean;
+}
+
+/**
+ * https://discord.com/developers/docs/interactions/slash-commands#applicationcommandoption
```
Comment (monbrey) [inline payloads/v9/_interactions/slashCommands.ts:54]:
```suggestion
 * https://discord.com/developers/docs/interactions/slash-commands#application-command-object-application-command-option-structure
```

File: payloads/v9/_interactions/slashCommands.ts [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+ * In contrast to `APIApplicationCommandSubCommandOptions`, these types cannot have an `options` array,
+ * but they can have a `choices` one
+ */
+export interface APIApplicationCommandArgumentOptions extends Omit<APIApplicationCommandOptionBase, 'type'> {
+	type: ApplicationCommandOptionType.String | ApplicationCommandOptionType.Integer;
+	choices?: APIApplicationCommandOptionChoice[];
+}
+
+/**
+ * https://discord.com/developers/docs/interactions/slash-commands#applicationcommandoptiontype
```
Comment (monbrey) [inline payloads/v9/_interactions/slashCommands.ts:83]:
```suggestion
 * https://discord.com/developers/docs/interactions/slash-commands#application-command-object-application-command-option-type
```

File: payloads/v9/_interactions/slashCommands.ts [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+	Integer,
+	Boolean,
+	User,
+	Channel,
+	Role,
+	Mentionable,
+}
+
+/**
+ * https://discord.com/developers/docs/interactions/slash-commands#applicationcommandoptionchoice
```
Comment (monbrey) [inline payloads/v9/_interactions/slashCommands.ts:98]:
```suggestion
 * https://discord.com/developers/docs/interactions/slash-commands#application-command-object-application-command-option-choice-structure
```

File: payloads/v9/_interactions/slashCommands.ts [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+ */
+export interface APIApplicationCommandOptionChoice {
+	name: string;
+	value: string | number;
+}
+
+/**
+ * https://discord.com/developers/docs/interactions/slash-commands#interaction-applicationcommandinteractiondata
```
Comment (monbrey) [inline payloads/v9/_interactions/slashCommands.ts:106]:
```suggestion
 * https://discord.com/developers/docs/interactions/slash-commands#interaction-object-interaction-structure
```

File: payloads/v9/_interactions/slashCommands.ts [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+
+/**
+ * https://discord.com/developers/docs/resources/guild#guild-member-object
+ */
+export interface APIInteractionDataResolvedGuildMember extends Omit<APIGuildMember, 'user' | 'deaf' | 'mute'> {
+	permissions: Permissions;
+}
+
+/**
+ * https://discord.com/developers/docs/interactions/slash-commands#interaction-applicationcommandinteractiondataoption
```
Comment (monbrey) [inline payloads/v9/_interactions/slashCommands.ts:135]:
```suggestion
 * https://discord.com/developers/docs/interactions/slash-commands#interaction-object-application-command-interaction-data-option-structure
```

File: payloads/v9/_interactions/slashCommands.ts [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+interface InteractionDataOptionBase<T extends ApplicationCommandOptionType, D = unknown> {
+	name: string;
+	type: T;
+	value: D;
+}
+
+// PERMISSIONS
+
+/**
+ * https://discord.com/developers/docs/interactions/slash-commands#guildapplicationcommandpermissions
```
Comment (monbrey) [inline payloads/v9/_interactions/slashCommands.ts:207]:
```suggestion
 * https://discord.com/developers/docs/interactions/slash-commands#application-command-permissions-object-guild-application-command-permissions-structure
```

File: payloads/v9/_interactions/slashCommands.ts [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+	 */
+	guild_id: Snowflake;
+	/**
+	 * The permissions for the command in the guild
+	 */
+	permissions: APIApplicationCommandPermission[];
+}
+
+/**
+ * https://discord.com/developers/docs/interactions/slash-commands#applicationcommandpermissions
```
Comment (monbrey) [inline payloads/v9/_interactions/slashCommands.ts:229]:
```suggestion
 * https://discord.com/developers/docs/interactions/slash-commands#application-command-permissions-object-application-command-permissions-structure
```

File: payloads/v9/channel.ts [hunk 11]
```diff
@@ -1043,85 +1051,107 @@ export const enum ComponentType {
...
+	 * The URL to direct users to when clicked for Link buttons
+	 */
+	url: string;
+}
+
+export type APIButtonComponent = APIButtonComponentWithCustomID | APIButtonComponentWithURL;
+
 /**
- * https://discord.com/developers/docs/interactions/message-components#component-object
+ * https://discord.com/developers/docs/interactions/message-components#buttons-button-styles
```
Comment (monbrey) [inline payloads/v9/channel.ts:1107]:
```suggestion
 * https://discord.com/developers/docs/interactions/message-components#button-object-button-styles
```

**Review by monbrey — CHANGES_REQUESTED** [review]

File: payloads/v8/channel.ts [hunk 12]
```diff
@@ -927,85 +931,107 @@ export const enum ComponentType {
...
+interface APIButtonComponentBase<Style extends ButtonStyle> extends APIBaseMessageComponent<ComponentType.Button> {
 	/**
 	 * The label to be displayed on the button
 	 */
 	label?: string;
-	/**
-	 * The custom_id to be sent in the interaction when clicked
-	 */
-	custom_id?: string;
+
```
Comment (SpaceEEC) [inline payloads/v8/channel.ts:951]: I only looked at a few files, but it doesn't seem to look like there were empty lines like these.
```suggestion
```

File: payloads/v8/channel.ts [hunk 13]
```diff
@@ -927,85 +931,107 @@ export const enum ComponentType {
...
+	style: Style;
 	/**
 	 * The emoji to display to the left of the text
 	 */
 	emoji?: APIPartialEmoji;
-	/**
-	 * The URL to direct users to when clicked for Link buttons
-	 */
-	url?: string;
+
```
Comment (SpaceEEC) [inline payloads/v8/channel.ts:960]:
```suggestion
```

File: payloads/v9/channel.ts [hunk 14]
```diff
@@ -1043,85 +1051,107 @@ export const enum ComponentType {
...
+interface APIButtonComponentBase<Style extends ButtonStyle> extends APIBaseMessageComponent<ComponentType.Button> {
 	/**
 	 * The label to be displayed on the button
 	 */
 	label?: string;
-	/**
-	 * The custom_id to be sent in the interaction when clicked
-	 */
-	custom_id?: string;
+
```
Comment (SpaceEEC) [inline payloads/v9/channel.ts:1071]:
```suggestion
```

File: payloads/v9/channel.ts [hunk 15]
```diff
@@ -1043,85 +1051,107 @@ export const enum ComponentType {
...
+	style: Style;
 	/**
 	 * The emoji to display to the left of the text
 	 */
 	emoji?: APIPartialEmoji;
-	/**
-	 * The URL to direct users to when clicked for Link buttons
-	 */
-	url?: string;
+
```
Comment (SpaceEEC) [inline payloads/v9/channel.ts:1080]:
```suggestion
```

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
I only have these nits and a comment, otherwise LGTM. 👀 
~~Although I didn't do throughout testing.~~

**Pull request merged**